src/tracking.F90

```diff
@@ -85,20 +119,121 @@ subroutine transport(p)
 
       if (p % material /= p % last_material) call calculate_xs(p)
 
+      ! Determine whether the particle is now transferred to correct domain.
+      ! Move particle forward the previous domain distance and calculating new
+      ! distance to domain boundary. If the total dd distance is still less than
+      ! stored tracking distance, let the particle cross domain once more.
+      ! Note the particle coordinates are recovered back.
+      if (dd_run .and. p % stored_distance > ZERO) then
+        xyz_temp = p % coord(1) % xyz
+        p % coord(1) % xyz = xyz_temp + p % fly_dd_distance * p % coord(1) % uvw
+        call distance_to_mesh_surface(p, domain_decomp % mesh, &
+             distance, d_dd_mesh, meshbin=domain_decomp % meshbin)
+        p % coord(1) % xyz = xyz_temp
```

Comment:
Rather than modifying `p % coord(1) % xyz` here in `transport`, I think we should just use a shifted `xyz` inside `distance_to_mesh_surface` itself.  Am I correct in thinking that `p % stored_distance` is exactly zero when we make the other call to `distance_to_mesh_surface`?  If so, then we can check `p % stored_distance` inside `distance_to_mesh_surface` to see if the coordinates need to be offset or not.